sdk/nodejs/tests/runtime/tsClosureCases.ts

```diff
@@ -1309,7 +1309,7 @@ return function () { console.log(v); };
 
 var __v = {};
 var __v_d1_proto = {};
-__f1.prototype = __v_d1_proto;
+Object.defineProperty(__f1, "prototype", { value: __v_d1_proto });
```

Comment:
The one thing I'm slightly nervous about is that this change will cause a lot of usage of closure serialization to change - which will lead to e.g. redeploying all Lambdas when this version of Pulumi is adopted. That's not necessarily a blocker - though it is a downside we'll need to message. 

I was initially wondering if this particular part of the change could be suppressed - but I do see that it is also technically incorrect to do what we were doing previously. 

How would you characterize the cases where serialization would change after this PR?  Is it "if you serialize a class definition, the serialization will be different"?

Reply:
> How would you characterize the cases where serialization would change after this PR? Is it "if you serialize a class definition, the serialization will be different"?

That characterization sounds good to me.

This PR could also affect any function whose property is defined via `Object.defineProperty`, but I don't think there are many use cases other than class definition.